src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorTagHelperBinderPhase.cs

```diff
@@ -350,13 +352,13 @@ public override void VisitRazorDirective(RazorDirectiveSyntax node)
 
             internal static bool IsTypeInNamespace(string typeName, string @namespace)
             {
-                if (!TrySplitNamespaceAndType(typeName, out var typeNamespace, out var _) || typeNamespace == string.Empty)
+                if (!TrySplitNamespaceAndType(typeName, out var typeNamespace, out var _) || typeNamespace.Length == 0)
```

Comment:
Why is `typeNamespace == string.Empty` bad? Does string.Empty allocate?

Reply:
You have to allocate the typeNamespace string before you can do that comparison

---
In reply to: [439617650](https://github.com/dotnet/aspnetcore/pull/22887#discussion_r439617650) [](ancestors = 439617650)

Reply:
Oh derp. Makes total sense 😄